send.go

```diff
@@ -198,6 +202,64 @@ func (cli *Client) SendMessage(ctx context.Context, to types.JID, message *waPro
 	}
 	resp.ID = req.ID
 
+	isInlineBotMode := false
+
+	if !req.InlineBotJID.IsEmpty() {
+		if !req.InlineBotJID.IsBot() {
+			err = ErrInvalidInlineBotID
+			return
+		}
+		isInlineBotMode = true
+	}
+
+	isBotMode := isInlineBotMode || to.IsBot()
+	var botNode = waBinary.Node{
+		Tag: "bot",
+	}
+
+	if isBotMode {
+		// common code for both inline and not inline modes
+		message.MessageContextInfo = &waE2E.MessageContextInfo{
+			BotMetadata: &waE2E.BotMetadata{
+				PersonaID: &req.BotPersonaID,
+			},
+			MessageSecret: random.Bytes(32),
+		}
+
+		if isInlineBotMode {
+			// inline mode specific code
+			messageSecret := message.GetMessageContextInfo().GetMessageSecret()
+			message = &waE2E.Message{
+				BotInvokeMessage: &waE2E.FutureProofMessage{
+					Message: &waE2E.Message{
+						ExtendedTextMessage: message.ExtendedTextMessage,
+						MessageContextInfo: &waE2E.MessageContextInfo{
+							BotMetadata: message.MessageContextInfo.BotMetadata,
+						},
+					},
+				},
+				MessageContextInfo: message.MessageContextInfo,
+			}
+
+			botMessage := &waE2E.Message{
+				BotInvokeMessage: message.BotInvokeMessage,
+				MessageContextInfo: &waE2E.MessageContextInfo{
+					BotMetadata:      message.MessageContextInfo.BotMetadata,
+					BotMessageSecret: applyBotMessageHKDF(messageSecret),
+				},
+			}
+
+			messagePlaintext, deviceSentMessagePlaintext, marshalErr := marshalMessage(req.InlineBotJID, botMessage)
+			if marshalErr != nil {
+				err = marshalErr
+				return
+			}
+
+			participantNodes, _ := cli.encryptMessageForDevices(ctx, []types.JID{req.InlineBotJID}, ownID, resp.ID, messagePlaintext, deviceSentMessagePlaintext, waBinary.Attrs{})
```

Comment:
The bot should never be the current user so no need to pass the device-sent wrapper through

```suggestion
			messagePlaintext, _, marshalErr := marshalMessage(req.InlineBotJID, botMessage)
			if marshalErr != nil {
				err = marshalErr
				return
			}

			participantNodes, _ := cli.encryptMessageForDevices(ctx, []types.JID{req.InlineBotJID}, ownID, resp.ID, messagePlaintext, nil, waBinary.Attrs{})
```

Reply:
alright, I was just keeping the standard